var/spack/repos/builtin/packages/atmi/package.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Atmi(CMakePackage):
+    """Asynchronous Task and Memory Interface, or ATMI, is a runtime framework
+       and programming model for heterogeneous CPU-GPU systems. It provides a
+       consistent, declarative API to create task graphs on CPUs and GPUs
+       (integrated and discrete)."""
+
+    homepage = "https://github.com/RadeonOpenCompute/atmi"
+    url      = "https://github.com/RadeonOpenCompute/atmi/archive/rocm-3.5.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+
+    version('3.5.0', sha256='3fb57d2e583fab82bd0582d0c2bccff059ca91122c18ac49a7770a8bb041a37b')
+
+    variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
+    depends_on('cmake@3.5.2', type='build')
+    depends_on('comgr@3.5:', type='build', when='@3.5:')
+    depends_on('hsa-rocr-dev@3.5:', type='build', when='@3.5:')
+    depends_on('libelf@0.8:', type='build', when='@3.5:')
```

Comment:
`when='@3.5:'` isn't needed here since you don't have any versions older than 3.5. Also, are you sure these are all build dependencies? Build deps are added to the `PATH` during installation, but a lot of these looks like libraries that should be linked to. The default dep type (build/link) should be sufficient, or you can explicitly specify link if it doesn't need to be in the `PATH`.

Reply:
when='@3.5:' was added based on need to support future releases too. As you may find atmi for rocm-3.5.0 is dependent on rocm-3.5.0 versions of comgr, hsa-rocr-dev etc . Whereas atmi for rocm-3.8.0 will be dependent on rocm-3.8.0 versions of comgr ,hsa-rocr-dev. 
My understanding is that for rocm3.8.0, we will be adding the below in addition to what we have currently
     depends_on('comgr@3.8:', type='build', when='@3.8:')
    depends_on('hsa-rocr-dev@3.8:', type='build', when='@3.8:')
The build deps were added mainly because atmi is dependent on comgr and hsa-rocr-dev. If they are not built already, those are built initially and then atmi will be built.

Reply:
In that case, a nice way to handle this might be:
```suggestion
    for ver in ['3.5.0', '3.8.0', ...]:
        depends_on('comgr@' + ver, type='build', when='@' + ver)
        depends_on('hsa-rocr-dev@ver', type='build', when='@' + ver)
    depends_on('libelf@0.8:', type='build')
```
Does romc 3.5.0 require exactly comgr 3.5.0, or can it build with 3.5.0+? If it's the former, the above should be correct, if it's the latter, add a trailing colon to the version number to represent +.

Reply:
yes, rcom 3.5.0 require exactly comgr 3.5.0 etc